include/secp256k1.h

```diff
@@ -169,6 +169,17 @@ typedef int (*secp256k1_nonce_function)(
 #  define SECP256K1_ARG_NONNULL(_x)
 # endif
 
+/** Attribute for marking functions, types, and variables as deprecated */
+#if !defined(SECP256K1_BUILD) && defined(__has_attribute)
```

Comment:
IIRC GCC doesn't have `__has_attribute` until very recent versions, but does have `__attribute__((__deprecated__))` since at least GCC 4.1 (tested with godbolt) and probably earlier. Do we perhaps want this additionally enabled on based on `__GNUC__`?

Reply:
This is what I had earlier but @jonasnick  pointed out that already GCC >= 5 offers `__has__attribute`. Moreover, `__attribute__((__deprecated__))` indeed works in >= GCC 4.1 but the optional `msg` argument works only in GCC >= 4.5 (in fact you get an error if you pass an argument in GCC < 4.5. 

(See the resolved discussion above).

Reply:
Oh wow, I thought __has_attribute was only added much later. Looks good then.